pkg/smi/client.go

```diff
@@ -288,6 +303,40 @@ func (c *client) ListTrafficTargets(options ...TrafficTargetListOption) []*smiAc
 	return trafficTargets
 }
 
+func isValidTrafficTarget(trafficTarget *smiAccess.TrafficTarget) bool {
+	// destination namespace must be same as traffic target namespace
+	if trafficTarget.Namespace != trafficTarget.Spec.Destination.Namespace {
+		return false
+	}
+
+	if !hasValidRules(trafficTarget.Spec.Rules) {
+		return false
+	}
+
+	return true
+}
+
+// hasValidRules checks if the given SMI TrafficTarget object has valid rules
+func hasValidRules(rules []smiAccess.TrafficTargetRule) bool {
+	if rules == nil {
+		return false
+	}
+	if len(rules) == 0 {
+		return false
+	}
```

Comment:
I was curious if this could be collapsed. (Not proposing we do - just curious.)
It looks like `len(rules)` could cover both cases: https://play.golang.org/p/LzR_HKKywFj
![image](https://user-images.githubusercontent.com/49918230/135183262-c51b8b9d-ee4d-4b40-b245-9c2d9dc70fd2.png)